baybe/kernels/basic.py

```diff
@@ -13,6 +13,62 @@
 from baybe.utils.validation import finite_float
 
 
+@define(frozen=True)
+class CosineKernel(Kernel):
+    """A cosine kernel."""
+
+    period_length_prior: Optional[Prior] = field(
+        default=None, validator=optional_v(instance_of(Prior))
+    )
+    """An optional prior on the kernel periodic length."""
+
+    period_length_initial_value: Optional[float] = field(
+        default=None, converter=optional_c(float), validator=optional_v(finite_float)
+    )
+    """An optional initial value for the kernel periodic length."""
+
+    def to_gpytorch(self, *args, **kwargs):  # noqa: D102
+        # See base class.
+        import torch
+
+        from baybe.utils.torch import DTypeFloatTorch
+
+        gpytorch_kernel = super().to_gpytorch(*args, **kwargs)
+        if (initial_value := self.period_length_initial_value) is not None:
+            gpytorch_kernel.period_length = torch.tensor(
+                initial_value, dtype=DTypeFloatTorch
+            )
+        return gpytorch_kernel
+
+
+@define(frozen=True)
+class LinearKernel(Kernel):
+    """A linear kernel."""
+
+    variance_prior: Optional[Prior] = field(
+        default=None, validator=optional_v(instance_of(Prior))
+    )
+    """An optional prior on the variance parameter."""
+
+    variance_initial_value: Optional[float] = field(
```

Comment:
We have the exact same field definition all over the place. It could make sense to define it upfront a la `optional_finite_float = field(...)` and then reuse it in the classes? (An even better solution would be custom types here which I've already asked for on attrs Github but which is still pending .... will try another push there)

Reply:
1) Is this actually saving much? We would loose that coders would directly see the detailed characteristic of the atttribute in place
2) have you tested this with sphinx whether it still works as intended?

Reply:
I guess it should work with sphinx. But if you don't like it, we can also skip it. I'd anyway rather prefer the type-based approach (just pinged the attrs guys again)